main/views.py

```diff
@@ -1,102 +1,115 @@
 """All views are here."""
-from django.shortcuts import render, get_object_or_404, redirect
-from .models import Post, Comment
-from .forms import PostForm, CommentForm
 from django.contrib.auth.decorators import login_required
+from django.utils.decorators import method_decorator
+from django.shortcuts import get_object_or_404, redirect
+from django.core.urlresolvers import reverse_lazy
+from django.views.generic import DetailView, ListView, CreateView, UpdateView, View, DeleteView, TemplateView
+from .forms import CommentForm
+from .models import Post, Comment
+
+
+class Protected(View):
+    """Protect views that need to show only for authorised users."""
+
+    @method_decorator(login_required)
+    def dispatch(self, *args, **kwargs):
+        """Rewrite standart method and it to decorator."""
+        return super(Protected, self).dispatch(*args, **kwargs)
```

Comment:
`super`'s arguments may be omitted in Python3 and you should pass `self` to dispatch.